android/app/src/main/java/com/mattermost/helpers/Credentials.java

```diff
@@ -1,26 +1,79 @@
 package com.mattermost.helpers;
 
 import androidx.annotation.Nullable;
+import androidx.biometric.BiometricPrompt;
+import androidx.core.content.ContextCompat;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.WritableMap;
 import com.oblador.keychain.KeychainModule;
+import com.oblador.keychain.exceptions.KeyStoreAccessException;
 
+import java.util.concurrent.Executor;
 
 public class Credentials {
+
+    private static void authenticateUserWithBiometrics(ReactApplicationContext context, BiometricPrompt.AuthenticationCallback callback) {
+        Executor executor = ContextCompat.getMainExecutor(context);
+        BiometricPrompt biometricPrompt = new BiometricPrompt(
+                context.getCurrentActivity(),
+                executor,
+                callback
+        );
+
+        BiometricPrompt.PromptInfo promptInfo = new BiometricPrompt.PromptInfo.Builder()
+                .setTitle("Authenticate to retrieve secret")
+                .setSubtitle("Log in using your biometric credential")
+                .setNegativeButtonText("Use account password")
+                .build();
+
+        try {
+            Cipher cipher = getCipher(); // Your logic to initialize and return the Cipher
+            BiometricPrompt.CryptoObject cryptoObject = new BiometricPrompt.CryptoObject(cipher);
+            biometricPrompt.authenticate(promptInfo, cryptoObject);
+        } catch (Exception e) {
+            callback.onAuthenticationError(BiometricPrompt.ERROR_NO_BIOMETRICS, "Failed to initialize Cipher: " + e.getMessage());
+        }
+    }
+
+    private static Cipher getCipher() throws Exception {
+        // Your logic to initialize and return the Cipher
+        // For example:
+        // KeyStore keyStore = KeyStore.getInstance("AndroidKeyStore");
+        // keyStore.load(null);
+        // Key key = keyStore.getKey("your-key-alias", null);
+        // Cipher cipher = Cipher.getInstance("AES/CBC/PKCS7Padding");
+        // cipher.init(Cipher.ENCRYPT_MODE, key);
+        // return cipher;
+    }
+
     public static void getCredentialsForServer(ReactApplicationContext context, String serverUrl, ResolvePromise promise) {
-        final KeychainModule keychainModule = new KeychainModule(context);
+        authenticateUserWithBiometrics(context, new BiometricPrompt.AuthenticationCallback() {
```

Comment:
This does not seem optional, also irrc some functions from push notifications do make use of this function as well as createThumbnail for videos and I'm not sure this does not disrupt the current flow.